### PR TITLE
Add macos DMG building using cpack 

### PR DESCRIPTION
**Pull request by kdheepak**
Addresses #379 

Run the following from the build folder to create the DMG 

```
mkdir build-osx 
cd build-osx
cmake -DENABLE_PACKAGE_BUILD=ON -DCPACK_BINARY_BUNDLE=ON -DBUILD_PYTHON_INTERFACE=ON -DCMAKE_INSTALL_PREFIX="/Users/$USER/local/helics-develop/" ../
make -j 8 && make install
cpack -C Release
```

### CODE REVIEW

**Comment by kdheepak** [comment]
One needs to copy `Info.plist` and `HELICS.ico` to the build folder, before running `cpack -C Release`. This can be done through CMake in the configuration step.

**Comment by kdheepak** [comment]
If we can compile the shared library linking the static library for ZMQ, Boost and the standard library that would be ideal. Users would need to download just one file and everything would ideally just work.

**Comment by phlptp** [comment]
@kdheepak 
do you know the commands for compiling the standard library statically in clang.  I think the current code is missing that piece.  it woulds on Windows and gcc currently.  But if we knew what was required we might be able to tweak that for APPLE as well

**Comment by phlptp** [comment]
if we need to do the copy step that probably should be done in cmake rather than leave it up to the users

File: CMakeLists.txt [hunk 1]
```diff
@@ -532,6 +532,11 @@ else (WIN32)
 	set(CPACK_PACKAGE_ICON "${CMAKE_SOURCE_DIR}/docs/img/HELICS.ico")
 endif(WIN32)
 
+if (APPLE)
+    set(CPACK_BUNDLE_NAME "libhelics")
+    set(CPACK_BUNDLE_PLIST Info.plist)
+    set(CPACK_BUNDLE_ICON HELICS.ico)
```
Comment (phlptp) [inline CMakeLists.txt]: can these be set to the full path instead of needing to copy them over?

Comment (kdheepak) [reply CMakeLists.txt]: Yes, I will do that.

**Comment by kdheepak** [comment]
This worked! I can create a dmg now using static ZMQ (that builds from source). I have the flag set for static Boost and static standard library, but I don't know if that is actually doing anything at the moment.

File: config/Info.plist [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	<key>CFBundleInfoDictionaryVersion</key>
+	<string>1.0</string>
+	<key>CFBundlePackageType</key>
+	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>%VERS%</string>
+	<key>CFBundleVersion</key>
+	<string>%VERS%</string>
+	<key>NSHumanReadableCopyright</key>
+	<string>© 2017 HELICS</string>
```
Comment (phlptp) [inline config/Info.plist]: should this be 2018?

File: config/Info.plist [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	<key>NSPrincipalClass</key>
+	<string>NSApplication</string>
+</dict>
+</plist>
```
Comment (phlptp) [inline config/Info.plist]: Is there anything in this file that is likely to change over time (other than the copyright once a year)

**Comment by phlptp** [comment]
what compiler are you using?

**Comment by kdheepak** [comment]
I'm using Clang. I still need to figure out what what all the fields in the plist file should be.

**Comment by phlptp** [comment]
my guess is you are not linking to the static c++ libraries in that case.

**Pull request merged**